Compiler crash with `assertion failed: asTerm called on not-a-Term val <none>`

## Compiler version

Reliably crashes with 3.8.2, 3.8.3, 3.8.0.
Works fine with 3.7.4 (compiler complains about missing methods in `InMemoryStorage` but rightly so)

## Minimized code

```Scala
import cats.data.EitherT
import cats.effect.IO
import scodec.bits.ByteVector
import fs2.Stream

import java.time.{Duration, Instant}

enum Expiration {
  case Never, Default
  case Deadline (val value: Instant)
  case Interval (val value: Duration)
}

trait Blob {
  def address: ByteVector
  def data: Stream[IO, Byte]
  def copy(expiration: Expiration = Expiration.Default, owner: Option[String] = None): IO[Blob]
  def delete(): IO[Unit]
}

trait Storage {
  def resolve(address: ByteVector): EitherT[IO, Throwable, Blob]
  def create(data: Stream[IO, Byte], expiration: Expiration = Expiration.Default, owner: Option[String] = None): IO[Blob]
}

final class InMemoryStorage extends Storage {

  import InMemoryStorage.*

  // Yes, this class is incomplete, we are missing the methods from
  // the Storage trait. But that should not cause the compiler to crash
}

object InMemoryStorage {

  def apply(): InMemoryStorage = new InMemoryStorage
}
```

## Output (click arrow to expand)
<!--
Note: to get a full crash log when using scala-cli, run compilation with the flag `--server=false`
-->
<details>

```
scala: 
  unhandled exception while running MegaPhase{crossVersionChecks, firstTransform, checkReentrant, elimPackagePrefixes, cookComments, checkLoopingImplicits, betaReduce, inlineVals, expandSAMs, elimRepeated, refchecks, dropForMap} on /Users/dirk/Development/Scala/Archive/archive-storage/src/main/scala/archive/storage/blobs/surface.scala

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose
  For non-enriched exceptions, compile with -Xno-enrich-error-messages.


     while compiling: /Users/dirk/Development/Scala/Archive/archive-storage/src/main/scala/archive/storage/blobs/surface.scala
        during phase: MegaPhase{crossVersionChecks, firstTransform, checkReentrant, elimPackagePrefixes, cookComments, checkLoopingImplicits, betaReduce, inlineVals, expandSAMs, elimRepeated, refchecks, dropForMap}
                mode: Mode(ImplicitsEnabled)
     library version: version 3.8.2
    compiler version: version 3.8.2
            settings: -classpath /Users/dirk/Development/Scala/Archive/archive-storage/target/scala-3.8.2/classes:/Users/dirk/Development/Scala/Archive/archive-kernel/target/scala-3.8.2/classes:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-core_3/2.13.0/cats-core_3-2.13.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect_3/3.7.0/cats-effect_3-3.7.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/co/fs2/fs2-core_3/3.13.0/fs2-core_3-3.13.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/co/fs2/fs2-io_3/3.13.0/fs2-io_3-3.13.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scodec/scodec-core_3/2.3.3/scodec-core_3-2.3.3.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect-testing-scalatest_3/1.8.0/cats-effect-testing-scalatest_3-1.8.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest_3/3.2.20/scalatest_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-kernel_3/2.13.0/cats-kernel_3-2.13.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.3.7/scala3-library_3-3.3.7.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect-kernel_3/3.7.0/cats-effect-kernel_3-3.7.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect-std_3/3.7.0/cats-effect-std_3-3.7.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-mtl_3/1.6.0/cats-mtl_3-1.6.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scodec/scodec-bits_3/1.2.4/scodec-bits_3-1.2.4.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/comcast/ip4s-core_3/3.8.0/ip4s-core_3-3.8.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect-testing-core_3/1.8.0/cats-effect-testing-core_3-1.8.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-core_3/3.2.20/scalatest-core_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-featurespec_3/3.2.20/scalatest-featurespec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-flatspec_3/3.2.20/scalatest-flatspec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-freespec_3/3.2.20/scalatest-freespec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-funsuite_3/3.2.20/scalatest-funsuite_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-funspec_3/3.2.20/scalatest-funspec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-propspec_3/3.2.20/scalatest-propspec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-refspec_3/3.2.20/scalatest-refspec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-wordspec_3/3.2.20/scalatest-wordspec_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-diagrams_3/3.2.20/scalatest-diagrams_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-matchers-core_3/3.2.20/scalatest-matchers-core_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-shouldmatchers_3/3.2.20/scalatest-shouldmatchers_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-mustmatchers_3/3.2.20/scalatest-mustmatchers_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.16/scala-library-2.13.16.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/literally_3/1.2.0/literally_3-1.2.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/idna4s-core_3/0.1.0/idna4s-core_3-0.1.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalactic/scalactic_3/3.2.20/scalactic_3-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scalatest/scalatest-compatible/3.2.20/scalatest-compatible-3.2.20.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_3/2.1.0/scala-xml_3-2.1.0.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-collections-core_3/0.9.10/cats-collections-core_3-0.9.10.jar:/Users/dirk/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/algebra_3/2.13.0/algebra_3-2.13.0.jar -d /Users/dirk/Development/Scala/Archive/archive-storage/target/scala-3.8.2/classes

scala: ## Exception when compiling 1 sources to /Users/dirk/Development/Scala/Archive/archive-storage/target/scala-3.8.2/classes
java.lang.AssertionError: assertion failed: asTerm called on not-a-Term val <none>
scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
dotty.tools.dotc.core.Symbols$Symbol.asTerm(Symbols.scala:186)
dotty.tools.dotc.ast.tpd$.New(tpd.scala:573)
dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:315)
dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:307)
dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:301)
dotty.tools.dotc.core.Types$MethodTypeCompanion.addAnnotation$1(Types.scala:4350)
dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4360)
dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4364)
dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2$$anonfun$1(Types.scala:4341)
scala.collection.immutable.List.map(List.scala:240)
dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2(Types.scala:4341)
dotty.tools.dotc.core.Types$MethodType.<init>(Types.scala:4262)
dotty.tools.dotc.core.Types$CachedMethodType.<init>(Types.scala:4287)
dotty.tools.dotc.core.Types$MethodTypeCompanion.apply(Types.scala:4367)
dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols(Types.scala:4342)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:891)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef$$anonfun$1(Scala2Unpickler.scala:990)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef(Scala2Unpickler.scala:990)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:901)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.$anonfun$6(Scala2Unpickler.scala:630)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.parseToCompletion$1(Scala2Unpickler.scala:630)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete$$anonfun$1(Scala2Unpickler.scala:677)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:20)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete(Scala2Unpickler.scala:675)
dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:175)
dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform(DenotTransformers.scala:44)
dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform$(DenotTransformers.scala:36)
dotty.tools.dotc.transform.ElimRepeated.transform(ElimRepeated.scala:65)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:840)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
dotty.tools.dotc.core.Symbols$Symbol.recomputeDenot(Symbols.scala:124)
dotty.tools.dotc.core.Symbols$Symbol.computeDenot(Symbols.scala:118)
dotty.tools.dotc.core.Symbols$Symbol.denot(Symbols.scala:109)
dotty.tools.dotc.core.Symbols$Symbol.name(Symbols.scala:277)
dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames$$anonfun$2(SymDenotations.scala:2420)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.IterableOnceOps.foreach(IterableOnce.scala:633)
scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:337)
scala.collection.AbstractIterator.foreach(Iterator.scala:1328)
dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames(SymDenotations.scala:2420)
dotty.tools.dotc.core.SymDenotations$MemberNamesImpl.apply(SymDenotations.scala:3006)
dotty.tools.dotc.core.SymDenotations$ClassDenotation.memberNames(SymDenotations.scala:2402)
dotty.tools.dotc.core.Types$Type.memberNames(Types.scala:1050)
dotty.tools.dotc.core.Types$Type.memberDenots(Types.scala:1067)
dotty.tools.dotc.core.Types$Type.typeAliasMembers(Types.scala:1138)
dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix$lzyINIT1(PlainPrinter.scala:164)
dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix(PlainPrinter.scala:164)
dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:245)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:322)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
dotty.tools.dotc.cc.Capabilities$CoreCapability.toText(Capability.scala:65)
dotty.tools.dotc.cc.Capabilities$CoreCapability.toText$(Capability.scala:64)
dotty.tools.dotc.core.Types$TypeRef.toText(Types.scala:3000)
dotty.tools.dotc.printing.Printer.toTextLocal$$anonfun$1(Printer.scala:184)
dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
dotty.tools.dotc.printing.Printer.toTextLocal(Printer.scala:184)
dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:259)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:294)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
dotty.tools.dotc.core.Types$Type.toText(Types.scala:2102)
dotty.tools.dotc.printing.Printer.toTextGlobal$$anonfun$1(Printer.scala:188)
dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
dotty.tools.dotc.printing.Printer.toTextGlobal(Printer.scala:188)
dotty.tools.dotc.printing.PlainPrinter.toTextRHS$$anonfun$1(PlainPrinter.scala:593)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:595)
dotty.tools.dotc.printing.PlainPrinter.paramText$1(PlainPrinter.scala:394)
dotty.tools.dotc.printing.PlainPrinter.paramsText$$anonfun$1(PlainPrinter.scala:395)
scala.collection.immutable.List.map(List.scala:240)
dotty.tools.dotc.printing.PlainPrinter.paramsText(PlainPrinter.scala:395)
dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1$$anonfun$5(PlainPrinter.scala:309)
dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
dotty.tools.dotc.printing.Printer.changePrec(Printer.scala:73)
dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:313)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:364)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
dotty.tools.dotc.core.Types$Type.toText(Types.scala:2102)
dotty.tools.dotc.printing.Printer.toTextGlobal$$anonfun$1(Printer.scala:188)
dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
dotty.tools.dotc.printing.Printer.toTextGlobal(Printer.scala:188)
dotty.tools.dotc.printing.PlainPrinter.toTextRHS$$anonfun$1(PlainPrinter.scala:584)
dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:595)
dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:528)
dotty.tools.dotc.printing.PlainPrinter.dclTextWithInfo(PlainPrinter.scala:663)
dotty.tools.dotc.printing.PlainPrinter.dclText(PlainPrinter.scala:659)
dotty.tools.dotc.core.Denotations$SingleDenotation.showDcl(Denotations.scala:999)
dotty.tools.dotc.typer.RefChecks$.membersStrings$1$$anonfun$2(RefChecks.scala:738)
scala.collection.immutable.List.map(List.scala:240)
dotty.tools.dotc.typer.RefChecks$.membersStrings$1(RefChecks.scala:738)
dotty.tools.dotc.typer.RefChecks$.stubImplementations$1(RefChecks.scala:741)
dotty.tools.dotc.typer.RefChecks$.checkNoAbstractMembers$1(RefChecks.scala:757)
dotty.tools.dotc.typer.RefChecks$.checkAllOverrides(RefChecks.scala:927)
dotty.tools.dotc.typer.RefChecks.transformTemplate(RefChecks.scala:1379)
dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1071)
dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:377)
dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:272)
dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:452)
dotty.tools.dotc.transform.MegaPhase.loop$1(MegaPhase.scala:465)
dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:465)
dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:396)
dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:399)
dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:481)
dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:493)
dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:406)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.immutable.List.foreach(List.scala:327)
dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:399)
dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:380)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1324)
dotty.tools.dotc.Run.runPhases$1(Run.scala:373)
dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$2(Run.scala:420)
dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$adapted$1(Run.scala:420)
scala.Function0.apply$mcV$sp(Function0.scala:42)
dotty.tools.dotc.Run.showProgress(Run.scala:482)
dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:420)
dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:432)
dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
dotty.tools.dotc.Run.compileUnits(Run.scala:432)
dotty.tools.dotc.Run.compileSources(Run.scala:319)
dotty.tools.dotc.Run.compile(Run.scala:304)
dotty.tools.dotc.Driver.doCompile(Driver.scala:38)
dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:141)
dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:196)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:252)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:186)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:166)
sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:241)
sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:166)
sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:214)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:542)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:542)
sbt.internal.inc.Incremental$.$anonfun$apply$3(Incremental.scala:182)
sbt.internal.inc.Incremental$.$anonfun$apply$3$adapted(Incremental.scala:180)
sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:458)
sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:264)
sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:413)
sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:500)
sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
sbt.internal.inc.Incremental$.apply(Incremental.scala:208)
sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:542)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:496)
sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:433)
sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
org.jetbrains.jps.incremental.scala.local.SbtCompiler.$anonfun$doCompile$3(SbtCompiler.scala:87)
scala.util.Try$.apply(Try.scala:217)
org.jetbrains.jps.incremental.scala.local.SbtCompiler.doCompile(SbtCompiler.scala:85)
org.jetbrains.jps.incremental.scala.local.SbtCompiler.compile(SbtCompiler.scala:17)
org.jetbrains.jps.incremental.scala.local.LocalServer.doCompile(LocalServer.scala:47)
org.jetbrains.jps.incremental.scala.local.LocalServer.compile(LocalServer.scala:26)
org.jetbrains.jps.incremental.scala.remote.Main$.compileLogic(Main.scala:213)
org.jetbrains.jps.incremental.scala.remote.Main$.$anonfun$handleCommand$1(Main.scala:194)
org.jetbrains.jps.incremental.scala.remote.Main$.decorated$1(Main.scala:181)
org.jetbrains.jps.incremental.scala.remote.Main$.handleCommand(Main.scala:191)
org.jetbrains.jps.incremental.scala.remote.Main$.serverLogic(Main.scala:164)
org.jetbrains.jps.incremental.scala.remote.Main$.nailMain(Main.scala:104)
org.jetbrains.jps.incremental.scala.remote.Main.nailMain(Main.scala)
java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
java.base/java.lang.reflect.Method.invoke(Method.java:565)
com.facebook.nailgun.NGSession.runImpl(NGSession.java:312)
com.facebook.nailgun.NGSession.run(NGSession.java:198)
           
scala: Compilation failed when compiling to: /Users/dirk/Development/Scala/Archive/archive-storage/target/scala-3.8.2/classes
  assertion failed: asTerm called on not-a-Term val <none>
  scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
  dotty.tools.dotc.core.Symbols$Symbol.asTerm(Symbols.scala:186)
  dotty.tools.dotc.ast.tpd$.New(tpd.scala:573)
  dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:315)
  dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:307)
  dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:301)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.addAnnotation$1(Types.scala:4350)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4360)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4364)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2$$anonfun$1(Types.scala:4341)
  scala.collection.immutable.List.map(List.scala:240)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2(Types.scala:4341)
  dotty.tools.dotc.core.Types$MethodType.<init>(Types.scala:4262)
  dotty.tools.dotc.core.Types$CachedMethodType.<init>(Types.scala:4287)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.apply(Types.scala:4367)
  dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols(Types.scala:4342)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:891)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef$$anonfun$1(Scala2Unpickler.scala:990)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef(Scala2Unpickler.scala:990)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:901)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.$anonfun$6(Scala2Unpickler.scala:630)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.parseToCompletion$1(Scala2Unpickler.scala:630)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete$$anonfun$1(Scala2Unpickler.scala:677)
  scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:20)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
  dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete(Scala2Unpickler.scala:675)
  dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:175)
  dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
  dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
  dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform(DenotTransformers.scala:44)
  dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform$(DenotTransformers.scala:36)
  dotty.tools.dotc.transform.ElimRepeated.transform(ElimRepeated.scala:65)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:840)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:857)
  dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:886)
  dotty.tools.dotc.core.Symbols$Symbol.recomputeDenot(Symbols.scala:124)
  dotty.tools.dotc.core.Symbols$Symbol.computeDenot(Symbols.scala:118)
  dotty.tools.dotc.core.Symbols$Symbol.denot(Symbols.scala:109)
  dotty.tools.dotc.core.Symbols$Symbol.name(Symbols.scala:277)
  dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames$$anonfun$2(SymDenotations.scala:2420)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
  scala.collection.IterableOnceOps.foreach(IterableOnce.scala:633)
  scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:337)
  scala.collection.AbstractIterator.foreach(Iterator.scala:1328)
  dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames(SymDenotations.scala:2420)
  dotty.tools.dotc.core.SymDenotations$MemberNamesImpl.apply(SymDenotations.scala:3006)
  dotty.tools.dotc.core.SymDenotations$ClassDenotation.memberNames(SymDenotations.scala:2402)
  dotty.tools.dotc.core.Types$Type.memberNames(Types.scala:1050)
  dotty.tools.dotc.core.Types$Type.memberDenots(Types.scala:1067)
  dotty.tools.dotc.core.Types$Type.typeAliasMembers(Types.scala:1138)
  dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix$lzyINIT1(PlainPrinter.scala:164)
  dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix(PlainPrinter.scala:164)
  dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:245)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
  dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:322)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
  dotty.tools.dotc.cc.Capabilities$CoreCapability.toText(Capability.scala:65)
  dotty.tools.dotc.cc.Capabilities$CoreCapability.toText$(Capability.scala:64)
  dotty.tools.dotc.core.Types$TypeRef.toText(Types.scala:3000)
  dotty.tools.dotc.printing.Printer.toTextLocal$$anonfun$1(Printer.scala:184)
  dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
  dotty.tools.dotc.printing.Printer.toTextLocal(Printer.scala:184)
  dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:259)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
  dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:294)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
  dotty.tools.dotc.core.Types$Type.toText(Types.scala:2102)
  dotty.tools.dotc.printing.Printer.toTextGlobal$$anonfun$1(Printer.scala:188)
  dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
  dotty.tools.dotc.printing.Printer.toTextGlobal(Printer.scala:188)
  dotty.tools.dotc.printing.PlainPrinter.toTextRHS$$anonfun$1(PlainPrinter.scala:593)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:595)
  dotty.tools.dotc.printing.PlainPrinter.paramText$1(PlainPrinter.scala:394)
  dotty.tools.dotc.printing.PlainPrinter.paramsText$$anonfun$1(PlainPrinter.scala:395)
  scala.collection.immutable.List.map(List.scala:240)
  dotty.tools.dotc.printing.PlainPrinter.paramsText(PlainPrinter.scala:395)
  dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1$$anonfun$5(PlainPrinter.scala:309)
  dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
  dotty.tools.dotc.printing.Printer.changePrec(Printer.scala:73)
  dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:313)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
  dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:364)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:366)
  dotty.tools.dotc.core.Types$Type.toText(Types.scala:2102)
  dotty.tools.dotc.printing.Printer.toTextGlobal$$anonfun$1(Printer.scala:188)
  dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
  dotty.tools.dotc.printing.Printer.toTextGlobal(Printer.scala:188)
  dotty.tools.dotc.printing.PlainPrinter.toTextRHS$$anonfun$1(PlainPrinter.scala:584)
  dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
  dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
  dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:595)
  dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:528)
  dotty.tools.dotc.printing.PlainPrinter.dclTextWithInfo(PlainPrinter.scala:663)
  dotty.tools.dotc.printing.PlainPrinter.dclText(PlainPrinter.scala:659)
  dotty.tools.dotc.core.Denotations$SingleDenotation.showDcl(Denotations.scala:999)
  dotty.tools.dotc.typer.RefChecks$.membersStrings$1$$anonfun$2(RefChecks.scala:738)
  scala.collection.immutable.List.map(List.scala:240)
  dotty.tools.dotc.typer.RefChecks$.membersStrings$1(RefChecks.scala:738)
  dotty.tools.dotc.typer.RefChecks$.stubImplementations$1(RefChecks.scala:741)
  dotty.tools.dotc.typer.RefChecks$.checkNoAbstractMembers$1(RefChecks.scala:757)
  dotty.tools.dotc.typer.RefChecks$.checkAllOverrides(RefChecks.scala:927)
  dotty.tools.dotc.typer.RefChecks.transformTemplate(RefChecks.scala:1379)
  dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1071)
  dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
  dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
  dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
  dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:377)
  dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
  dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:272)
  dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:452)
  dotty.tools.dotc.transform.MegaPhase.loop$1(MegaPhase.scala:465)
  dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:465)
  dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:396)
  dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:399)
  dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
  dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:481)
  dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:493)
  dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:406)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
  scala.collection.immutable.List.foreach(List.scala:327)
  dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:399)
  dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:380)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
  scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
  scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1324)
  dotty.tools.dotc.Run.runPhases$1(Run.scala:373)
  dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$2(Run.scala:420)
  dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$adapted$1(Run.scala:420)
  scala.Function0.apply$mcV$sp(Function0.scala:42)
  dotty.tools.dotc.Run.showProgress(Run.scala:482)
  dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:420)
  dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:432)
  dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
  dotty.tools.dotc.Run.compileUnits(Run.scala:432)
  dotty.tools.dotc.Run.compileSources(Run.scala:319)
  dotty.tools.dotc.Run.compile(Run.scala:304)
  dotty.tools.dotc.Driver.doCompile(Driver.scala:38)
  dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:141)
  dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
  sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
  sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:196)
  scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
  sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:252)
  sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:186)
  sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:166)
  sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:241)
  sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:166)
  sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:214)
  sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:542)
  sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:542)
  sbt.internal.inc.Incremental$.$anonfun$apply$3(Incremental.scala:182)
  sbt.internal.inc.Incremental$.$anonfun$apply$3$adapted(Incremental.scala:180)
  sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:458)
  sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
  sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
  sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
  sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:264)
  sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:413)
  sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:500)
  sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
  sbt.internal.inc.Incremental$.apply(Incremental.scala:208)
  sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:542)
  sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:496)
  sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
  sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:433)
  sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
  org.jetbrains.jps.incremental.scala.local.SbtCompiler.$anonfun$doCompile$3(SbtCompiler.scala:87)
  scala.util.Try$.apply(Try.scala:217)
  org.jetbrains.jps.incremental.scala.local.SbtCompiler.doCompile(SbtCompiler.scala:85)
  org.jetbrains.jps.incremental.scala.local.SbtCompiler.compile(SbtCompiler.scala:17)
  org.jetbrains.jps.incremental.scala.local.LocalServer.doCompile(LocalServer.scala:47)
  org.jetbrains.jps.incremental.scala.local.LocalServer.compile(LocalServer.scala:26)
  org.jetbrains.jps.incremental.scala.remote.Main$.compileLogic(Main.scala:213)
  org.jetbrains.jps.incremental.scala.remote.Main$.$anonfun$handleCommand$1(Main.scala:194)
  org.jetbrains.jps.incremental.scala.remote.Main$.decorated$1(Main.scala:181)
  org.jetbrains.jps.incremental.scala.remote.Main$.handleCommand(Main.scala:191)
  org.jetbrains.jps.incremental.scala.remote.Main$.serverLogic(Main.scala:164)
  org.jetbrains.jps.incremental.scala.remote.Main$.nailMain(Main.scala:104)
  org.jetbrains.jps.incremental.scala.remote.Main.nailMain(Main.scala)
  java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  java.base/java.lang.reflect.Method.invoke(Method.java:565)
  com.facebook.nailgun.NGSession.runImpl(NGSession.java:312)
  com.facebook.nailgun.NGSession.run(NGSession.java:198)
Errors occurred while compiling module 'Archive.archive-storage.main'
Finished, saving caches…
Builder Scala sbt builder requested build stop
Executing post-compile tasks…
Synchronizing output directories…
03.04.26, 09:53 - Build completed with 2 errors and 1 warning in 1 sec, 935 ms
```
</details>

I've got a minimization but it seems quite artificial:

build.sh:
```shell
#!/usr/bin/env bash
set -euo pipefail

SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
mkdir -p "$SCRIPT_DIR/.out/classes"

# Old scala3-library (3.3.7): caps was a package object, no scala.caps.internal.consume
OLD_STDLIB_CP=$(cs fetch "org.scala-lang:scala3-library_3:3.3.7" --classpath 2>/dev/null)

# Compile using Scala 3.8.0 compiler with only the stale 3.3.7 stdlib on the classpath.
# Crucially: NO scala-library-3.8.x.jar, so scala.caps.internal.consume is NOT found.
cs launch "org.scala-lang:scala3-compiler_3:3.8.0" \
   --main-class dotty.tools.dotc.Main \
   -- \
   -classpath "$OLD_STDLIB_CP" \
   -d "$SCRIPT_DIR/.out/classes" \
   "$SCRIPT_DIR/repro.scala"
```

repro.scala:
```scala
class Foo { def bar(): Unit }
```

It seems impossible to reproduce this using  scala-cli because it adds the proper stdlib to the classpath automatically

<details>

  unhandled exception while running MegaPhase{crossVersionChecks, firstTransform, checkReentrant, elimPackagePrefixes, cookComments, checkLoopingImplicits, betaReduce, inlineVals, expandSAMs, elimRepeated, refchecks, dropForMap} on /Users/mpalka/Projects/scala3/local/repro/i25702/source/repro.scala

  An unhandled exception was thrown in the compiler.
  Please file a crash report here:
  https://github.com/scala/scala3/issues/new/choose
  For non-enriched exceptions, compile with -Xno-enrich-error-messages.


     while compiling: /Users/mpalka/Projects/scala3/local/repro/i25702/source/repro.scala
        during phase: MegaPhase{crossVersionChecks, firstTransform, checkReentrant, elimPackagePrefixes, cookComments, checkLoopingImplicits, betaReduce, inlineVals, expandSAMs, elimRepeated, refchecks, dropForMap}
                mode: Mode(ImplicitsEnabled)
     library version: version 3.8.0
    compiler version: version 3.8.0
            settings: -classpath /Users/mpalka/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.3.7/scala3-library_3-3.3.7.jar:/Users/mpalka/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.16/scala-library-2.13.16.jar -d /Users/mpalka/Projects/scala3/local/repro/i25702/out/classes

Exception in thread "main" java.lang.AssertionError: assertion failed: asTerm called on not-a-Term val <none>
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:10)
	at dotty.tools.dotc.core.Symbols$Symbol.asTerm(Symbols.scala:186)
	at dotty.tools.dotc.ast.tpd$.New(tpd.scala:564)
	at dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:257)
	at dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:251)
	at dotty.tools.dotc.core.Annotations$Annotation$.apply(Annotations.scala:245)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.addAnnotation$1(Types.scala:4328)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4338)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.adaptParamInfo(Types.scala:4342)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2$$anonfun$1(Types.scala:4319)
	at scala.collection.immutable.List.map(List.scala:240)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols$$anonfun$2(Types.scala:4319)
	at dotty.tools.dotc.core.Types$MethodType.<init>(Types.scala:4240)
	at dotty.tools.dotc.core.Types$CachedMethodType.<init>(Types.scala:4265)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.apply(Types.scala:4345)
	at dotty.tools.dotc.core.Types$MethodTypeCompanion.fromSymbols(Types.scala:4320)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:891)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef$$anonfun$1(Scala2Unpickler.scala:990)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readTypeRef(Scala2Unpickler.scala:990)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.readType(Scala2Unpickler.scala:901)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.$anonfun$6(Scala2Unpickler.scala:630)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.at(Scala2Unpickler.scala:309)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.parseToCompletion$1(Scala2Unpickler.scala:630)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete$$anonfun$1(Scala2Unpickler.scala:677)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:20)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler.atReadPos(Scala2Unpickler.scala:319)
	at dotty.tools.dotc.core.unpickleScala2.Scala2Unpickler$LocalUnpickler.complete(Scala2Unpickler.scala:675)
	at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:175)
	at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
	at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
	at dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform(DenotTransformers.scala:44)
	at dotty.tools.dotc.core.DenotTransformers$InfoTransformer.transform$(DenotTransformers.scala:36)
	at dotty.tools.dotc.transform.ElimRepeated.transform(ElimRepeated.scala:65)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:835)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:852)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:852)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:852)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:852)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.goForward$1(Denotations.scala:852)
	at dotty.tools.dotc.core.Denotations$SingleDenotation.current(Denotations.scala:881)
	at dotty.tools.dotc.core.Symbols$Symbol.recomputeDenot(Symbols.scala:124)
	at dotty.tools.dotc.core.Symbols$Symbol.computeDenot(Symbols.scala:118)
	at dotty.tools.dotc.core.Symbols$Symbol.denot(Symbols.scala:109)
	at dotty.tools.dotc.core.Symbols$Symbol.name(Symbols.scala:277)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames$$anonfun$2(SymDenotations.scala:2420)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:633)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:337)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1328)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.computeMemberNames(SymDenotations.scala:2420)
	at dotty.tools.dotc.core.SymDenotations$MemberNamesImpl.apply(SymDenotations.scala:3006)
	at dotty.tools.dotc.core.SymDenotations$ClassDenotation.memberNames(SymDenotations.scala:2402)
	at dotty.tools.dotc.core.Types$Type.memberNames(Types.scala:1048)
	at dotty.tools.dotc.core.Types$Type.memberDenots(Types.scala:1065)
	at dotty.tools.dotc.core.Types$Type.typeAliasMembers(Types.scala:1136)
	at dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix$lzyINIT1(PlainPrinter.scala:164)
	at dotty.tools.dotc.printing.PlainPrinter.printWithoutPrefix(PlainPrinter.scala:164)
	at dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:245)
	at dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
	at dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
	at dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
	at dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:322)
	at dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
	at dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
	at dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:364)
	at dotty.tools.dotc.printing.PlainPrinter.op$proxy1$1(PlainPrinter.scala:316)
	at dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1$$anonfun$6(PlainPrinter.scala:316)
	at dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
	at dotty.tools.dotc.printing.Printer.changePrec(Printer.scala:73)
	at dotty.tools.dotc.printing.PlainPrinter.toText$$anonfun$1(PlainPrinter.scala:317)
	at dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
	at dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
	at dotty.tools.dotc.printing.PlainPrinter.toText(PlainPrinter.scala:373)
	at dotty.tools.dotc.printing.RefinedPrinter.toText$$anonfun$1(RefinedPrinter.scala:362)
	at dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
	at dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
	at dotty.tools.dotc.printing.RefinedPrinter.toText(RefinedPrinter.scala:364)
	at dotty.tools.dotc.core.Types$Type.toText(Types.scala:2100)
	at dotty.tools.dotc.printing.Printer.toTextGlobal$$anonfun$1(Printer.scala:188)
	at dotty.tools.dotc.printing.Printer.atPrec(Printer.scala:45)
	at dotty.tools.dotc.printing.Printer.toTextGlobal(Printer.scala:188)
	at dotty.tools.dotc.printing.PlainPrinter.toTextRHS$$anonfun$1(PlainPrinter.scala:584)
	at dotty.tools.dotc.printing.MessageLimiter.controlled(MessageLimiter.scala:23)
	at dotty.tools.dotc.printing.PlainPrinter.controlled(PlainPrinter.scala:50)
	at dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:595)
	at dotty.tools.dotc.printing.PlainPrinter.toTextRHS(PlainPrinter.scala:528)
	at dotty.tools.dotc.printing.PlainPrinter.dclTextWithInfo(PlainPrinter.scala:663)
	at dotty.tools.dotc.printing.PlainPrinter.dclText(PlainPrinter.scala:657)
	at dotty.tools.dotc.core.Symbols$Symbol.showDcl(Symbols.scala:440)
	at dotty.tools.dotc.typer.RefChecks$.showDclAndLocation$1(RefChecks.scala:763)
	at dotty.tools.dotc.typer.RefChecks$.undefined$1(RefChecks.scala:765)
	at dotty.tools.dotc.typer.RefChecks$.checkNoAbstractMembers$1$$anonfun$2(RefChecks.scala:831)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:327)
	at dotty.tools.dotc.typer.RefChecks$.checkNoAbstractMembers$1(RefChecks.scala:761)
	at dotty.tools.dotc.typer.RefChecks$.checkAllOverrides(RefChecks.scala:927)
	at dotty.tools.dotc.typer.RefChecks.transformTemplate(RefChecks.scala:1379)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1071)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1072)
	at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:377)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
	at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:272)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:452)
	at dotty.tools.dotc.transform.MegaPhase.loop$1(MegaPhase.scala:465)
	at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:465)
	at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:396)
	at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:399)
	at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:454)
	at dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:481)
	at dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:493)
	at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:386)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:327)
	at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:379)
	at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:380)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1327)
	at dotty.tools.dotc.Run.runPhases$1(Run.scala:373)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$2(Run.scala:420)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$adapted$1(Run.scala:420)
	at scala.Function0.apply$mcV$sp(Function0.scala:45)
	at dotty.tools.dotc.Run.showProgress(Run.scala:482)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:420)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:432)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:432)
	at dotty.tools.dotc.Run.compileSources(Run.scala:319)
	at dotty.tools.dotc.Run.compile(Run.scala:304)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:38)
	at dotty.tools.dotc.Driver.process(Driver.scala:213)
	at dotty.tools.dotc.Driver.process(Driver.scala:181)
	at dotty.tools.dotc.Driver.process(Driver.scala:193)
	at dotty.tools.dotc.Driver.main(Driver.scala:223)
	at dotty.tools.dotc.Main.main(Main.scala)
</details>

--- a/compiler/src/dotty/tools/dotc/core/Annotations.scala
+++ b/compiler/src/dotty/tools/dotc/core/Annotations.scala
@@ -26,7 +26,11 @@ object Annotations {
 
     def hasSymbol(sym: Symbol)(using Context) = symbol == sym
 
-    def matches(cls: Symbol)(using Context): Boolean = symbol.derivesFrom(cls)
+    def matches(cls: Symbol)(using Context): Boolean =
+      // No annotation matches `AnyClass`. This is necessary since `AnyClass`
+      // is returned if a requiredClass call fails, and we want to be
+      // conservative in this case.
+      (cls ne defn.AnyClass) && symbol.derivesFrom(cls)
 
     def appliesToModule: Boolean = true // for now; see remark in SymDenotations
 

--- a/tests/neg/i25702.scala
+++ b/tests/neg/i25702.scala
@@ -1,0 +1,14 @@
+// https://github.com/scala/scala3/issues/25702
+// Regression test: a concrete class missing trait methods must produce a clean
+// diagnostic, not a compiler crash. The original crash needed a stale
+// `scala3-library` where `scala.caps.internal.consume` was missing, which made
+// `requiredClass` fall back to `defn.AnyClass`. We can't reproduce the
+// stale-classpath setup in the standard test framework, but the underlying fix
+// is exercised by the `RequiredClassTest` unit test; this file guards the
+// "missing-method" diagnostic.
+
+trait Storage:
+  def resolve(address: String): String
+  def create(data: List[Byte], owner: Option[String] = None): String
+
+final class InMemoryStorage extends Storage // error

